src/jit/lsraarm.cpp

```diff
@@ -330,10 +330,17 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
                 var_types srcType = castOp->TypeGet();
                 emitAttr  cmpSize = EA_ATTR(genTypeSize(srcType));
 
-                // If we cannot store the comparisons in an immediate for either
-                // comparing against the max or min value, then we will need to
-                // reserve a temporary register.
+                // If we cannot store data in an immediate for instructions,
+                // then we will need to reserve a temporary register.
 
+                // For checking the type
+                bool canStoreTypeMask = emitter::emitIns_valid_imm_for_alu(castInfo.typeMask);
+                if (!canStoreTypeMask)
+                {
+                    info->internalIntCount = 1;
+                }
+
```

Comment:
It seems like the conditions here for specifying a temporary register don't quite replicate the conditions for using it in codegen. E.g., it seems this should be under
```
if (castInfo.unsignedSource || castInfo.unsignedDest)
```
or, better:
```
if (castInfo.signCheckOnly)
{
  // no temp needed
} else if (castInfo.unsignedSource || castInfo.unsignedDest)
  // check typeMask
} else {
  // check max/min value
}
```